libs/chain-signatures/contract/src/primitives/participants.rs

```diff
@@ -15,6 +19,55 @@ pub struct ParticipantInfo {
     pub sign_pk: PublicKey,
 }
 
+#[near(serializers=[borsh, json])]
+#[derive(PartialEq, Eq, PartialOrd, Ord, Clone, Default)]
+pub struct TeeParticipantInfo {
+    /// TEE Remote Attestation Quote that proves the participant's identity.
+    pub tee_quote: Vec<u8>,
+    /// Supplemental data for the TEE quote, including Intel certificates to verify it came from
+    /// genuine Intel hardware, along with details about the Trusted Computing Base (TCB)
+    /// versioning, status, and other relevant info.
+    pub quote_collateral: String,
+}
+
+/// Without this, the following tests fail with HostError(TotalLogLengthExceeded { length: 31510, limit: 16384 }):
+/// - tests::test_signature_simple
+/// - tests::test_signature_simple_legacy
+/// - tests::test_signature_timeout
+impl fmt::Debug for TeeParticipantInfo {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        fn hex_preview(bytes: &[u8], max: usize) -> String {
+            if bytes.len() <= max {
+                format!("{:02x?}", bytes)
+            } else {
+                format!("{:02x?}… ({} bytes)", &bytes[..max], bytes.len())
+            }
+        }
+        fn str_preview(s: &str, max: usize) -> String {
+            if s.len() <= max {
+                format!("{:?}", s)
+            } else {
+                format!("{:?}… ({} chars)", &s[..max], s.len())
+            }
+        }
+        f.debug_struct("TeeParticipantInfo")
+            .field("tee_quote", &hex_preview(&self.tee_quote, 256))
+            .field(
+                "quote_collateral",
+                &str_preview(&self.quote_collateral, 256),
+            )
+            .finish()
+    }
+}
+
+impl TeeParticipantInfo {
+    pub fn verify_quote(&self, timestamp: u64) -> Result<VerifiedReport, Error> {
+        let tee_collateral = get_collateral(self.quote_collateral.clone());
```

Comment:
so all `get_collateral` seems to be doing, is to de-compose the string.
https://github.com/Near-One/mpc/blob/5f86401ac055be3b8dae74365cf87749833d38cb/libs/chain-signatures/contract/src/primitives/tee/quote.rs#L45

Why don't we just have the nodes send a `QuoteCollateral` struct and use serde json serialization / deserialization?

And then store that struct in the contract instead of the `quote_collateral` string?